Wikipedia/View Controllers/WebView/WebViewController.m

```diff
@@ -257,6 +261,33 @@ - (void)viewDidLoad {
     self.webView.scrollView.layer.anchorPoint = CGPointMake((isRTL ? 1.0 : 0.0), 0.0);
 
     [self tocUpdateViewLayout];
+
+    [self.KVOControllerNonRetaining observe:self.webView.scrollView keyPath:WMF_SAFE_KEYPATH(UIScrollView.new, contentSize) options:NSKeyValueObservingOptionNew block:^(WebViewController* observer, id object, NSDictionary* change) {
+        [self.sectionTitlesViewController updateOverlaysPositions];
+        // Restrict the web view from scrolling horizonally.
+        [object preventHorizontalScrolling];
```

Comment:
why do we need to do this here? do we ever want it to scroll horizontally? if not, we might also want to consider handling cases when (after loading) the webview's content size is wider than the view's bounds.